test/functional/finalization_deposit_reorg.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Unit-e developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import UnitETestFramework
+from test_framework.util import (
+    connect_nodes,
+    sync_blocks,
+    assert_equal,
+    disconnect_nodes,
+    wait_until,
+)
+import os
+
+ESPERANZA_CONFIG = '-esperanzaconfig={"epochLength": 10}'
+
+
+class EsperanzaDepositReorgTest(UnitETestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+
+        self.extra_args = [
+            [ESPERANZA_CONFIG],
+            [ESPERANZA_CONFIG],
+            [ESPERANZA_CONFIG],
+            [ESPERANZA_CONFIG, '-validating=1'],
+        ]
+        self.setup_clean_chain = True
+
+    def setup_network(self):
+        # topology:
+        # node[0] -> node[1]
+        #         -> node[2]
+        #         -> node[3]
+        self.setup_nodes()
+        connect_nodes(self.nodes[0], self.nodes[1].index)
+        connect_nodes(self.nodes[0], self.nodes[2].index)
+        connect_nodes(self.nodes[0], self.nodes[3].index)
```

Comment:
`connect_nodes(self.nodes[0], self.nodes[3].index)`
Looks a bit weird to me. Such call would make a lot of sense if `self.nodes[3]` had a name...

Reply:
I give names to every node in `run_test`. 
Since `setup_network` is a different function than `run_test` I didn't create local variables there as they can become out of sync between functions once we touch/modify this test (e.g., we rename `node0` to `fork0`).

Do you suggest giving the name particularly to `self.nodes[3]` and name it `finalizer`?

Reply:
Not really. I suggest this:
`connect_nodes(self.nodes[0], 3)`

Reply:
There is of course another option. After `self.setup_nodes()` nodes are already created and you can assign them to fields for example. This will handle your issue with renaming
```python
def setup_network(self):
  self.setup_nodes()
...
  self.finalizer = self.nodes[3]
...
  connect_nodes(self.node0, self.finalizer.index)
```

Reply:
Thanks, will change!

Reply:
Changed to the first proposed solution.